src/main/java/org/capstone/job_fair/services/impl/job_fair/JobFairServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+import org.capstone.job_fair.services.mappers.JobFairMapper;
+import org.capstone.job_fair.utils.MessageUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class JobFairServiceImpl implements JobFairService {
+    @Autowired
+    JobFairRepository jobFairRepository;
+    @Autowired
+    AccountRepository accountRepository;
+    @Autowired
+    private JobFairMapper jobFairMapper;
+
+    @Override
+    public void createJobFair(JobFairDTO dto) {
+        dto.setStatus(JobFairStatus.DRAFT);
+        UserDetailsImpl userDetails = (UserDetailsImpl) SecurityContextHolder.getContext()
+                .getAuthentication().getPrincipal();
+        dto.setCreatorId(userDetails.getId());
+        if(dto.getCompanyBuyBoothEndTime() < dto.getCompanyBuyBoothStartTime())
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.END_TIME_LESS_THAN_START_TIME_ERROR));
+        if(dto.getCompanyRegisterEndTime() < dto.getCompanyRegisterStartTime())
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.END_TIME_LESS_THAN_START_TIME_ERROR));
+        if(dto.getEndTime() < dto.getStartTime())
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.END_TIME_LESS_THAN_START_TIME_ERROR));
+        if(dto.getEndTime() - dto.getStartTime() > DataConstraint.JobFair.VALID_TIME)
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.INVALID_END_TIME));
+        if(dto.getCompanyBuyBoothEndTime() - dto.getCompanyBuyBoothStartTime() > DataConstraint.JobFair.VALID_TIME)
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.INVALID_END_TIME));
+        if(dto.getCompanyRegisterEndTime() - dto.getCompanyRegisterStartTime() > DataConstraint.JobFair.VALID_TIME)
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.JobFair.INVALID_END_TIME));
+        JobFairEntity entity = jobFairMapper.toJobFairEntity(dto);
+        jobFairRepository.save(entity);
+    }
+
+    @Override
+    public List<JobFairDTO> getAllJobFairPlanOfEmployee() {
+        UserDetailsImpl userDetails = (UserDetailsImpl) SecurityContextHolder.getContext()
+                .getAuthentication().getPrincipal();
+        String id = userDetails.getId();
+       Optional<AccountEntity> account = accountRepository.findById(id);
+        if(!account.isPresent()) throw new IllegalArgumentException(
+                MessageUtil.getMessage(MessageConstant.Account.NOT_FOUND));
+        System.out.println(account.get().getRole().toString());
+        if(!account.get().getRole().getName().equals(Role.STAFF.toString())) throw new
```

Comment:
Do not check role here since in the controller we have already use @PreAuthorize to check for role